src/BenchmarkDotNet/Detectors/CpuDetector.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Linq;
+using BenchmarkDotNet.Detectors.Cpu;
+using BenchmarkDotNet.Detectors.Cpu.Linux;
+using BenchmarkDotNet.Detectors.Cpu.macOS;
+using BenchmarkDotNet.Detectors.Cpu.Windows;
+using BenchmarkDotNet.Extensions;
+using Perfolizer.Phd.Dto;
+
+namespace BenchmarkDotNet.Detectors;
+
+public class CpuDetector(params ICpuDetector[] detectors) : ICpuDetector
```

Comment:
similar to `ICpuDetector`, does it need to be public?

Reply:
From time to time, I check if the given version of BenchmarkDotNet is capable of detecting the CPU (or other environment properties) correctly in the given environment. Right now, I use `HostEnvironmentInfo.GetInformation()`, which is not very convenient. Also, I want to make features like CPU detection reusable. We have many cool detection features in the library, but when I need them somewhere else, I have to copy-paste the code and next maintain such a pseudo-fork. It would be much better if I could just reference `BenchmarkDotNet` and call the relevant method. Again, we can discuss how the API should look, but I definitely want to have a public function for CPU detection.

Reply:
FWIW I am using `--info` for that:

https://github.com/dotnet/BenchmarkDotNet/blob/ca5dfdf106affb0922f5b8002a67272a27d3fd03/src/BenchmarkDotNet/ConsoleArguments/CommandLineOptions.cs#L155-L156

> We have many cool detection features in the library

I see. In such a case we should consider moving such reusable helpers to a standalone package, so the users don't need to reference BenchmarkDotNet.dll itself as it's quite big.